src/wallet/wallet_rpc_server_commands_defs.h

```diff
@@ -313,13 +315,17 @@ namespace wallet_rpc
       std::list<std::string> amount_keys;
       uint64_t fee;
       std::string tx_blob;
+      std::string tx_metadata;
+      bool tx_metadata_appended;
```

Comment:
I do not understand these response flags. If the `tx_metadata` is NOT empty, that should imply that it was appended. This is necessary because `tx_metadata`_sometimes_ mirrors `tx_blob`, which is unnecessary.

Reply:
tx_metadata_appended means that the data in tx_metadata is only the suffix, with tx_blob being the prefix. This is to avoid sending a copy of tx_blob inside tx_metadata. Admittedly, it may be a bit weird as an api...

Reply:
After a second thought, I’m also getting confused (despite having approved). So this code stores the tx blob part and the suffix metadata part (which together form `pending_tx`) separately into `tx_blob` and `tx_metadata`, respectively, when `get_tx_hex` and `get_tx_metadata` are both set. Does this mean that a user making the `relay_tx` wallet RPC command needs to concatenate these two strings manually as the `hex` parameter? Isn’t that tedious/error-prone?

Is the sole purpose of this appended-or-not thing to avoid duplication of the same data in two fields? Does that redundancy really need to be avoided?

Or maybe, the `get_tx_hex` flag and `get_tx_metadata` flag could be made mutually exclusive, since their intended use cases seem completely orthogonal; i.e. the former is for pushing the tx blob directly to a daemon, while the latter is for letting the wallet handle the actual commit of a tx sometime later than the `transfer` command.

Reply:
Yes, the strings have to be concatenated if both outputs are requested. It might be error prone, but certainly not tedious since it's all automated. Unless you do RPC via curl and do all that manually, but that's certainly not the intent of RPC.

Redundancy is the only reason I did this. If it's deemed to not be a problem, I can remove it. It felt wasteful, but then I suppose it's just one single tx so may not be that bad after all. So if you think it's best to remove, I'll do that.

The flags are mutually exclusive only if you use them that way. I would not think someone wanting to commit a tx later would never want to have the complete tx too.

Reply:
Thanks. I'm fine with what it is now.

Reply:
Crap I missed this before it got merged ... sorry, I think this format is going to cause future maintenance headaches. Primarily because there are now additional field(s) to indicate how to parse other fields. I stated in another comment - there are two flags set by the RPC client already. The client can set the tx flag to false to reduce redundant information sent. And older clients will already have the metadata flag set to false anyway. I think its a win just to remove that `std::vector<bool>` now being sent in the RPC.

I'm going to open a PR and start a discussion specifically on that topic. On a related note, these binary-hex fields are a pain if you do not want to use the monero C++ libs directly, because the format is ill-specified.